docs/scopes.md

```diff
@@ -0,0 +1,199 @@
+
+This document defines the scope values
+accepted in the Firefox Accounts ecosystem,
+and the rules for parsing and validating them.
+
+## Short-name scope values
+
+FxA supports a small set of "short-name" scope values
+that are identified by a short English word.
+These correspond either to
+scope values defined by external specifications
+(e.g. OpenID Connect),
+or to legacy scope values
+introduced during early development.
+
+**No new short-name scope values should be added.**
+Instead we prefer to use URLs for new scope values,
+both to ensure uniqueness
+and to simplify parsing rules.
+
+Short-name scope values imply read-only access by default,
+with write access indicated by the suffix ":write".
+The may also have "sub-scopes"
+to indicate finer-grained access control.
+Each name component may contain only ascii alphanumeric characters
+and the underscore.
+
+For example:
+
+* `profile` indicates read-only access
+  to the user's profile data.
+* `profile:write` indicates read/write access
+  to the user's profile data.
+* `profile:display_name` indicates read-only access
+  to the user's display name, but not any other
+  profile data.
+* `profile:email:write` indicates read/write access
+  to the user's email address.
+
+The following short-name scope values are recognized
+in the FxA ecosystem.
+
+### Profile data
+
+* `profile`: access the user's profile data.
+* `profile:uid`: access the user's opaque user id.
+* `profile:email`: access the user's email address.
+* `profile:locale`: access the user's locale.
+* `profile:avatar`: access the user's avatar picture.
+* `profile:display_name`: access the user's human-readable display name.
+* `profile:amr`: access information about the user's authentication methods and 2FA status.
+
+### OpenID Connect
+
+* `openid`: used to request an OpenID Connect `id_token`.
+* `email`: a synonym for `profile:email`, defined by the OIDC spec.
+
+### OAuth Client Management
+
+* `clients`: access the list of OAuth clients connected to a user's account.
+* `oauth`: register a new OAuth client record.
+
+### Basket
+
+* `basket`: access the user's subscription data in
+  [basket](http://basket.readthedocs.io/)
+
+
+## URL Scopes
+
+For new capabilities, scope values are represented as URLs.
+This helps to ensure uniqueness
+and reduces ambiguity in parsing.
+URL-format scope value imply read/write access by default,
+are compared as heirarchical resource references,
+and use the hash fragment for permission qualifiers.
+For example:
+
+* `https://identity.mozilla.com/apps/oldsync` indicates full
+   access to the user's data in Firefox Sync.
+* `https://identity.mozilla.com/apps/oldsync/bookmarks` indicates
+   full access to the user's bookmark data in Firefox Sync,
+   but not to other data types.
+* `https://identity.mozilla.com/apps/oldsync#read` indicates
+   read-only access to the user's data in Firefox Sync.
+* `https://identity.mozilla.com/apps/oldsync/history#write` indicates
```

Comment:
Why would one want write-only access to history?

Reply:
If you're a mobile app that wants to record visits that can show up in the activity-stream on the user's desktop, but we don't want to let you read the sites that the user has been visiting on other devices.